### ISSUE TITLE
minimega: vm.setState race condition

### ISSUE THREAD

**Issue by jcrussell** [sandia-minimega/minimega#391]
We currently call `vm.setState` in launch from a deferred function because it needs to acquire the VM lock after launch or whatever releases it. This creates a race condition as another goroutine might also be waiting on the VM lock. We should make it so that `vm.setState` assumes that the caller already holds the lock. Perhaps we should also add a new function to BaseVM:

```
func (vm *BaseVM) setError(err error) {
    vm.tags["error"] = err.String()
    vm.setState(VM_ERROR)
}
```

This would be useful for debugging while VMs are in the error state without having to hunt through the logs.

**Comment by djfritz** [comment]
I'm for this. Doing this will resolve #388

**Comment by djfritz** [comment]
I like the error tag as well.

**Comment by jcrussell** [comment]
Cool -- I'm on it.
